src/Math-Matrix/PMTensorTest.class.st

```diff
@@ -0,0 +1,191 @@
+Class {
+	#name : #PMTensorTest,
+	#superclass : #TestCase,
+	#category : #'Math-Matrix'
+}
+
+{ #category : #tests }
+PMTensorTest >> testArray [
+
+	| t1 t2 |
+	t1 := PMTensor fromNestedArray: #( #( 1 2 3 4 ) #( 5 6 7 8 ) ).
+	self assert: t1 array equals: #( 1 2 3 4 5 6 7 8 ).
+
+	t2 := PMTensor fromNestedArray: #( #( #( 1 2 ) #( 3 4 ) ) #( #( 5 6 ) #( 7 8 ) )
+		         #( #( 9 10 ) #( 11 12 ) ) #( #( 13 14 ) #( 15 16 ) ) ).
+	self
+		assert: t2 array
+		equals: #( 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 )
+]
+
+{ #category : #tests }
+PMTensorTest >> testCreateScalarTensor [
+
+	| s |
+	s := PMTensor newWith: 2.
+	self assert: (s get: #(  )) equals: 2.
+	self should: [ s get: #( 1 1 ) ] raise: Error.
+	self assert: s rank equals: 0.
+	s set: #(  ) value: 1.
+	self assert: (s get: #(  )) equals: 1.
+	self assert: s shape equals: #(  ).
+	self assert: s size equals: 1
+]
+
+{ #category : #tests }
+PMTensorTest >> testFirst [
+
+	| a b |
+	a := PMTensor fromNestedArray: (1 to: 6) asArray.
+	self assert: a first equals: #( 1).
+	b := a reshape: #( 3 2 ).
+	self assert: b first equals: #( 1 1 )
+]
+
+{ #category : #tests }
+PMTensorTest >> testFlattenedIndexOf [
```

Comment:
The example values used in this test case are not well chosen, because they are `1 to: n` reshaped. This means that `flattenedIndexOf:` and `get:` return the same values, making is harder to understand what is going on.